src/indicators/flash-message.js

```diff
@@ -1,51 +1,59 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import { flashMessageType } from 'redux-flash'
 import classnames from 'classnames'
 
 /**
  *
- * A component that displays a flash message generated by [redux-flash](https://github.com/LaunchPadLab/redux-flash).
- * Any message props will be passed through to this component.
- * 
+ * A component that displays a flash message.
+ *
  * @name FlashMessage
  * @type Function
- * @param {Object} message - The flash message that will be displayed.
+ * @param {String} children - The flash message that will be displayed.
+ * @param {Boolean} [isError] - A flag to indicate whether the message is an error message.
  * @param {Function} [onDismiss] - A callback for dismissing the flash message. The dismiss button will only be shown if this callback is provided.
  * @example
- * 
- * function ManyMessages ({ messages }) {
+ *
+ * function MyView () {
+ *   const [message, setMessage] = useState(null)
  *   return (
  *      <div>
- *         {
- *            messages.map(message => <FlashMessage key={ message.id } message={ message } />)
- *         }
+ *        {
+ *           message &&
+ *           <FlashMessage>{message}</FlashMessage>)
+ *        }
+ *        <button onClick={() => setMessage('Hi!')}> Show message </button>
  *      </div>
  *   )
  * }
  *
  */
 
 const propTypes = {
-  message: flashMessageType.isRequired,
+  children: PropTypes.node.isRequired,
+  isError: PropTypes.bool,
   onDismiss: PropTypes.func,
+  className: PropTypes.string,
 }
 
 const defaultProps = {
+  isError: false,
   onDismiss: null,
+  className: '',
 }
 
-function FlashMessage ({ message, onDismiss }) {
-  const statusClass = message.isError ? 'failure' : 'success'
+function FlashMessage({ children, isError, onDismiss, className, ...rest }) {
+  const statusClass = isError ? 'failure' : 'success'
   return (
-    <div { ...message.props } className={ classnames('flash-message', statusClass, message.props.className) }>
-      { 
-        onDismiss &&
-        <button type="button" className="dismiss" onClick={ () => onDismiss(message) }>
+    <div
+      className={classnames('flash-message', statusClass, className)}
+      {...rest}
+    >
+      {onDismiss && (
+        <button type="button" className="dismiss" onClick={() => onDismiss()}>
           ×
         </button>
-      }
-      <p> { message.message } </p>
+      )}
+      <p> {children} </p>
```

Comment:
Now that we're accepting children, it's not necessarily clear to me that it can / should only accept phrasing content. In other words, should children be rendered in a `p` tag?

Reply:
Yeah, I thought about that. I decided to leave it in for backwards compat, but I can move it to `FlashMessageContainer` instead.